trunk/source/Library-Automata/src/de/uni_freiburg/informatik/ultimate/automata/petrinet/operations/DifferencePetriNet.java

```
@@ -581,56 +400,62 @@ public DifferenceSynchronizationInformation<LETTER, PLACE> computeDifferenceSync
-			ITransition<LETTER, PLACE> result = mInputTransition2State2OutputTransition.get(inputTransition, null);
+		private Transition<LETTER, PLACE>
+				getOrConstructTransitionCopy(final Transition<LETTER, PLACE> inputTransition) {
+			Transition<LETTER, PLACE> result = mInputTransition2State2OutputTransition.get(inputTransition, null);
 			if (result == null) {
-//				assert (mMinuendPlaces.containsAll(mMinued.getPredecessors(inputTransition))) : "missing predecessor";
 				final Set<PLACE> successors = new LinkedHashSet<>();
-				for (final PLACE petriNetSuccessor : mMinuend.getSuccessors(inputTransition)) {
+				for (final PLACE petriNetSuccessor : inputTransition.getSuccessors()) {
 					// possibly first time that we saw this place, add
 					addMinuendPlace(petriNetSuccessor);
 					successors.add(petriNetSuccessor);
 				}
 				final int totalOrderId = mNumberOfConstructedTransitions;
 				mNumberOfConstructedTransitions++;
 				result = mYetConstructedResult.addTransition(inputTransition.getSymbol(),
-						mMinuend.getPredecessors(inputTransition), ImmutableSet.of(successors), totalOrderId);
+						inputTransition.getPredecessors(), ImmutableSet.of(successors), totalOrderId);
 				mInputTransition2State2OutputTransition.put(inputTransition, null, result);
-				mTransitions.put(result, (Transition<LETTER, PLACE>) result);
-				final ITransition<LETTER, PLACE> valueBefore = mNew2Old.put(result, inputTransition);
+				final Transition<LETTER, PLACE> valueBefore = mNew2Old.put(result, inputTransition);
 				assert valueBefore == null : "Cannot add transition twice.";
 			}
 			return result;
 		}
 
 	}
 
+	@Override
+	public Set<Transition<LETTER, PLACE>> getSuccessors(final PLACE place) {
+		return mYetConstructedResult.getSuccessors(place);
+	}
+
+	@Override
+	public Set<Transition<LETTER, PLACE>> getPredecessors(final PLACE place) {
+		return mYetConstructedResult.getPredecessors(place);
+	}
+
```

Will this behave as expected? I.e., could it not be necessary to construct a place's successors (or perhaps predecessors) when this method is called (on-demand)?

---

I guess you are correct! The transitions are only added in `getOrConstructTransition(Transition<LETTER, PLACE>, PLACE)` that is only called by `getTransitions`. We should probably call it for all transitions with the corresponding place as a successor/predecessor.
Since it is not used (at least for now), we could also just throw an exception there.

---

Okay, I guess it is more complicated, since `getOrConstructTransition` is only a method of an inner class...
So we should rather throw an exception here.

---

See 5ecb9d5f500d43f5f2c2332b55e821961b69ebbd: Now there is an interface `IPetriNetTransitionProvider` (there is still the option to change the name...) between `IPetriNet` and `IPetriNetSuccessorProvider`. Therefore `DifferencePetriNet` does not need to implement these methods anymore.